gwpy/types/series.py

```diff
@@ -680,11 +680,11 @@ def _compare_index(self, other, axis="x"):
         except AttributeError:  # irregular index
             _index = "_{}index".format(axis)
             idxa = getattr(self, _index, None)
-            idxb = getattr(self, _index, None)
+            idxb = getattr(other, _index, None)
             if (
-                idxa is None  # index on 'other' but not on 'self'
-                or idxb is None  # index on 'self' but not on 'other'
-                or not numpy.array_equal(idxa, idxb)  # indexes don't match
+                idxa is None  # no index on 'self'
+                or idxb is None  # no index on 'other'
+                or numpy.array_equal(idxa, idxb)  # indexes are identical
```

Comment:
For series with a manual index, there isn't a clean concept of 'appending' or 'prepending' so I don't really like the idea that we are presuming that is the use case here. Purely for this function (which is independent of the prepend/append idea) I would like to keep the concept that equal indexes means 'compatible'.

Can you please revert the change on this one line? This may mean adding a different extra check in the relevant prepend/append function.